src/core/core.datasetController.js

```diff
@@ -971,15 +996,15 @@ export default class DatasetController {
 	_resyncElements() {
 		const me = this;
 		const numMeta = me._cachedMeta.data.length;
-		const numData = me._data.length;
+		const numData = me._cachedMeta._parsed.length;
 
 		if (numData > numMeta) {
 			me._insertElements(numMeta, numData - numMeta);
 		} else if (numData < numMeta) {
 			me._removeElements(numData, numMeta - numData);
 		}
 		// Re-parse the old elements (new elements are parsed in _insertElements)
-		me.parse(0, Math.min(numData, numMeta));
+		// me.parse(0, Math.min(numData, numMeta));
```

Comment:
I think we need to do the parsing first and resync the element counts after that. So compare `_data.length` to `_parsedRaw.length` and if that differs parse. then compare `_parsed.length` to `numMeta` and add the elements. need to remove the parsing from the `_insertElements.`